### ISSUE TITLE
[ci] [azure] [osx] Azure OSX build is broken due to gtk linking

### ISSUE THREAD

**Issue by ejgallego** [coq/coq#10501]
Since a few days the azure OSX workers do fail when trying to build coqide:
```
Undefined symbols for architecture x86_64:
  "_GTK_SOCKET", referenced from:
      _ml_gtk_socket_add_id in liblablgtk3_stubs.a(ml_gtk.o)
      _ml_gtk_socket_get_id in liblablgtk3_stubs.a(ml_gtk.o)
      _ml_gtk_socket_get_plug_window in liblablgtk3_stubs.a(ml_gtk.o)
OCAMLC    parsing/notation_gram.ml
  "_gtk_plug_get_type", referenced from:
      _ml_gtkwindow_init in liblablgtk3_stubs.a(ml_gtk.o)
  "_gtk_plug_new", referenced from:
      _ml_gtk_plug_new in liblablgtk3_stubs.a(ml_gtk.o)
     (maybe you meant: _ml_gtk_plug_new)
  "_gtk_socket_add_id", referenced from:
      _ml_gtk_socket_add_id in liblablgtk3_stubs.a(ml_gtk.o)
     (maybe you meant: _ml_gtk_socket_add_id)
  "_gtk_socket_get_id", referenced from:
      _ml_gtk_socket_get_id in liblablgtk3_stubs.a(ml_gtk.o)
     (maybe you meant: _ml_gtk_socket_get_id
)
  "_gtk_socket_get_plug_window", referenced from:
      _ml_gtk_socket_get_plug_window in liblablgtk3_stubs.a(ml_gtk.o)
     (maybe you meant: _ml_gtk_socket_get_plug_window)
  "_gtk_socket_get_type", referenced from:
      _ml_gtkwindow_init in liblablgtk3_stubs.a(ml_gtk.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
File "caml_startup", line 1:
Error: Error during linking
make[1]: *** [bin/coqide] Error 2
```
https://dev.azure.com/coq/coq/_build/results?buildId=4065

I am unsure if this is due to the OSX images now using `clang` instead of gcc, I guess an OSX expert is required here. Maybe trying with Dune could be interesting, but I'm not very optimisitic that will fix it.

This highlights a fatal flaw of our Azure setup which is that the builds are not reproducible and indeed we get upgrades randomly. That has been biting us quite a bit lately [on the other hand we discover issues pretty quickly :(]

**Comment by SkySkimmer** [comment]
It seems the brew `gtk+3` version from a success job was `3.24.9` and in a failing job it was `3.24.10`

**Comment by ejgallego** [comment]
> It seems the brew `gtk+3` version from a success job was `3.24.9` and in a failing job it was `3.24.10`

Seems strange tho that a minor release would produce this breakage tho, I remain inclined to thing that could be a clang / gcc update.

**Comment by SkySkimmer** [comment]
Could be related to https://github.com/Homebrew/homebrew-core/commit/9fd3fd9ed0e9ffa54417f3652c72be2bffe704c6 maybe

**Comment by SkySkimmer** [comment]
>Seems strange tho that a minor release would produce this breakage tho, I remain inclined to thing that could be a clang / gcc update.

Wouldn't they update the runner version when updating that stuff? (runner version is in "initialize job" section of the log, there was an update some time between https://dev.azure.com/coq/coq/_build/results?buildId=3578 (june 15) and https://dev.azure.com/coq/coq/_build/results?buildId=4035 (july 06), both of which passed, the july 06 one is the current version)

**Comment by vbgl** [comment]
Which C compiler is used in brew? in opam?

**Comment by vbgl** [comment]
Also, this linking error is quite expected:

> The GtkPlug and GtkSocket widgets are only available when GTK+ is compiled for the X11 platform

Source: https://developer.gnome.org/gtk3/stable/GtkPlug.html

**Comment by ejgallego** [comment]
Ummm why it worked before?

**Comment by vbgl** [comment]
I could reproduce the issue with gtk+3-3.24.10 (from homebrew). Compilation and linking work well when reinstalling gtk+3 at version 3.24.9 and reinstalling lablgtk3 and co.

**Comment by vbgl** [comment]
As a workaround (and maybe as a step towards more reproducibility on Azure — #10505), I propose that PR:  #10509

**Comment by SkySkimmer** [comment]
We now have a workaround, but how are we going to solve the issue in order to be able to update gtk eventually?

**Comment by vbgl** [comment]
This might be a general lablgtk3 issue that should be reported upstream.

**Comment by XVilka** [comment]
If it is an issue with lablgtk3, then it is better to report it now, because the first stable release is going to be done soon - https://github.com/garrigue/lablgtk/issues/90

**Comment by vbgl** [comment]
IIRC that was an issue of GTK+ at 3.24.10, fixed in 3.24.11

**Comment by ejgallego** [comment]
Indeed let's close then.
